crates/storage/src/address.rs

```diff
@@ -50,86 +51,52 @@ impl Id {
     /// ```
     ///
     #[must_use]
-    pub fn new() -> Self {
-        let mut bytes = [0; 16];
-        random_bytes(&mut bytes);
-        Self(uuid::Builder::from_random_bytes(bytes).into_uuid())
-    }
-
-    /// Returns a slice of 16 octets containing the value.
-    #[must_use]
-    pub const fn as_bytes(&self) -> &Bytes {
-        self.0.as_bytes()
+    pub fn new(bytes: [u8; 32]) -> Self {
+        // random_bytes(&mut bytes);
+        Self { bytes }
     }
 
     /// Root ID which is set to all zeroes by default.
     #[must_use]
-    pub const fn root() -> Self {
-        Self(Uuid::nil())
-    }
-}
-
-impl BorshDeserialize for Id {
-    fn deserialize(buf: &mut &[u8]) -> Result<Self, IoError> {
-        if buf.len() < 16 {
-            return Err(IoError::new(
-                IoErrorKind::UnexpectedEof,
-                "Not enough bytes to deserialize Id",
-            ));
-        }
-        let (bytes, rest) = buf.split_at(16);
-        *buf = rest;
-        Ok(Self(Uuid::from_slice(bytes).map_err(|err| {
-            IoError::new(IoErrorKind::InvalidData, err)
-        })?))
-    }
-
-    fn deserialize_reader<R: Read>(reader: &mut R) -> Result<Self, IoError> {
-        let mut bytes = [0_u8; 16];
-        reader.read_exact(&mut bytes)?;
-        Ok(Self(Uuid::from_bytes(bytes)))
+    pub fn root() -> Self {
+        Id::new([0; 32])
     }
-}
 
-impl BorshSerialize for Id {
-    fn serialize<W: Write>(&self, writer: &mut W) -> Result<(), IoError> {
-        writer.write_all(self.0.as_bytes())
+    #[must_use]
+    pub fn random() -> Id {
+        let mut bytes = [0_u8; 32];
+        random_bytes(&mut bytes);
+        Id::new(bytes)
     }
 }
 
 impl Default for Id {
     fn default() -> Self {
-        Self::new()
-    }
+            Self::new([0; 32])
+        }
 }
```

Comment:
let's take this out, and keep `Id` generation explicit, either through `new`, `root` or `random`